docs/_guide/styles.md

```diff
@@ -162,327 +122,653 @@ class MyElement extends LitElement {
-      :host { 
-        display: block; 
-        font-family: Roboto;
-        font-size: 20;
-        color: blue;
-      }
+      :host {
+        /* Warning: this approach has limitations & performance issues! */
+        color: ${myColor}
+      } 
     </style>
-    <p>Inherits font styles</p>
+    <div>template content</div>
   `;
 }
 ```
 
-{% include project.html folder="style/inherited" openFile="my-element.js" %}
+Expressions inside a `<style>` element won't update per instance in ShadyCSS, due to limitations of the ShadyCSS polyfill. See the [ShadyCSS readme](https://github.com/webcomponents/shadycss/blob/master/README.md#limitations) for more information.
+
+Additionally, evaluating an expression inside a `<style>` element is inefficient. When any text inside a `<style>` element changes, the browser must re-parse the whole `<style>` element, resulting in unnecessary work. 
+
+If you need to evaluate expressions inside a `<style>` element, use the following strategy to avoid creating performance problems:
+
+*   Separate styles that require per-instance evaluation from those that don't.
 
-If a host element itself inherits properties from another element, the host's shadow DOM children will inherit those properties in turn:
+*   Evaluate per-instance CSS properties by creating an expression that captures that property inside a complete `<style>` block. Include it in your template.
+
+**Example**
+
+```js
+{% include projects/style/perinstanceexpressions/my-element.js %}
+```
+
+{% include project.html folder="style/perinstanceexpressions" openFile="my-element.js" %}
+
+### Import an external stylesheet {#external-stylesheet}
+
+We recommend placing your styles in a static `styles` property for optimal performance. However, you can include an external style sheet in your template with a `<link>`:
+
+```js
+{% include projects/style/where/my-element.js %}
+```
+
+{% include project.html folder="style/where" openFile="my-element.js" %}
+
+There are some important caveats though:
+
+*  The [ShadyCSS polyfill](https://github.com/webcomponents/shadycss/blob/master/README.md#limitations) doesn't support external style sheets.
+
+*   External styles can cause a flash-of-unstyled-content (FOUC) while they load.
+
+*   The URL in the `href` attribute is relative to the **main document**. This is okay if you're building an app and your asset URLs are well-known, but avoid using external style sheets when building a reusable element.
+
+## Dynamic classes and styles
+
+One way to make styles dynamic is to add bindings to the `class` or `style` attributes in your template.
+
+The lit-html library offers two directives, `classMap` and `styleMap`, to conveniently apply classes and styles in HTML templates. 
+
+For more information on these and other lit-html directives, see the documentation on [lit-html built-in directives](https://lit-html.polymer-project.org/guide/template-reference#built-in-directives).
+
+To use `styleMap` and/or `classMap`:
```

Comment:
This feels a little repetitive, because there are dedicated sections about `styleMap` and `classMap` right below, with similar examples. How about moving the import statements directly into the examples below, and getting rid of this introductory example?